docs/ProgrammingConcerns.md

```diff
@@ -0,0 +1,170 @@
+# Please read this before contributing to FEX-Emu
+FEX-Emu is mostly a regular C++20 application but we have some interesting problems that most applications don't need to contend with.
+There are concerns with dealing with signals and memory allocation that we must be careful with handling.
+
+# Memory allocation routines
+## What is the problem?
+FEX-Emu needs to allocate memory differently than regular applications. This problem happens because FEX runs both 32-bit and 64-bit guest
+applications in the same address space as FEX itself. When running 32-bit applications, FEX reserves up all memory above 4GB in order to correctly
+emulate the 32-bit address space. We then use that reserved space for FEX allocations, so we don't interrupt application's own allocations.
+
+### Why not just just replace the system allocator?
+JEMalloc and similar allocators allow your application to replace the system allocator. This is something that FEX used to do, but this has since been
```

Comment:
Nit: The project name is spelt `jemalloc`.

Reply:
Fixed.